lib/operation.js

```diff
@@ -629,25 +629,37 @@ function boolean (operand, operator, options) {
 
 /**
  * Apply the linear formula a * input + b to the image (levels adjustment)
- * @param {number} [a=1.0] multiplier
- * @param {number} [b=0.0] offset
+ * @param {number | Array<number>} [a=[]] multiplier
+ * @param {number | Array<number>} [b=[]] offset
  * @returns {Sharp}
  * @throws {Error} Invalid parameters
  */
 function linear (a, b) {
+  if (!is.defined(a) && is.number(b)) {
```

Comment:
Would it make things easier/simpler if we change the API to enforce that both `a` and `b` are required, and are no longer optional? Calling `linear` without parameters is effectively a no-op anyway. (The type would still be `number | Array<number>`.)

Reply:
Up to you. It could be a breaking change for some dependent apps.

Reply:
I'm OK with breaking changes that simplify logic, but if it makes minimal difference to the complexity then let's keep the API as it is.